### DISCUSSION TITLE
m_forward_density_output and m_forward_rgb_output are both 16, why not 1 and 3 as the dimension.

### DISCUSSION THREAD

**Discussion by neilgogogo** [NVlabs/instant-ngp#363] [General]
Hi, I have a question that why the dimension of the m_forward_density_output and m_forward_rgb_output are both 16. As for the density, only the first element in the 16 elements is used and as for the rgb_output, only the first 3 elements are used every 16 elements. So, why not switch the dimension into 1 and 3 respectively to save the memory?

<img width="728" alt="image" src="https://user-images.githubusercontent.com/53148705/159205486-33b924d8-5871-4ed5-a067-d3c1555dc766.png">

<img width="595" alt="image" src="https://user-images.githubusercontent.com/53148705/159205499-8e91e2b9-eab0-4348-8957-664472b10ef1.png">

Many thanks.

**Comment by Tom94** [comment]
The hardware matrix multipliers of modern GPUs operate on blocks of data with fixed size. The underlying `FullyFusedMLP`, for example, uses 16x16x16 tensor core functionality. Ergo, outputs must be padded to 16 dimensions in order to use the acceleration.
